src/krb5/_localauth.pyx

```diff
@@ -0,0 +1,47 @@
+# Copyright: (c) 2022 Jordan Borean (@jborean93) <jborean93@gmail.com>
+# MIT License (see LICENSE or https://opensource.org/licenses/MIT)
+
+from libc.stdlib cimport free, malloc
+
+from krb5._exceptions import Krb5Error
+
+from krb5._context cimport Context
+from krb5._krb5_types cimport *
+from krb5._principal cimport Principal
+
+
+cdef extern from "python_krb5.h":
+
+    krb5_error_code krb5_aname_to_localname(
+        krb5_context context,
+        krb5_const_principal aname,
+        int lnsize,
+        char *lname_out
+    ) nogil
+
+    krb5_error_code KRB5_LNAME_NOTRANS
+
+def aname_to_localname(
+    context: Context,
+    principal: Principal
+) -> str:
```

Comment:
See comments about bytes, but this should also be `t.Optional[bytes]:` as it can return `None`.